### DISCUSSION TITLE
Platform to run

### DISCUSSION THREAD

**Discussion by Manzs1288** [edgexfoundry/community#317] [Q&A]
Hi, 
Is it possible to run edgexfoundry on virtual machine or embedded hardware like raspberry PI without docker container?

if it can be run only with container is it possible to make use of complete hardware level peripherals etc?

**Comment by cloudxxx8** [comment]
Yes, you just need to build your own binary for your platform
https://docs.edgexfoundry.org/3.1/getting-started/native/Ch-BuildRunNative/

**Reply by Manzs1288** [reply]
So you mean I can build and run directly on virtual machine and embedded hardware with and without any docker containers ?

**Reply by cloudxxx8** [reply]
yes

**Reply by cloudxxx8** [reply]
docker container is just a reference deployment

**Comment by qs-wang** [comment]
yes, you can run edgex on bare mental embeded arm box, docker, or k8s. 
I have tried run it on rockchip arm box.
